src/main/java/uk/gov/hmcts/reform/sscs/service/hmc/topic/ProcessHmcMessageService.java

```diff
@@ -37,6 +45,106 @@ public class ProcessHmcMessageService {
     private final CcdCaseService ccdCaseService;
 
     private final HearingUpdateService hearingUpdateService;
+    private final UpdateCcdCaseService updateCcdCaseService;
+    private final IdamService idamService;
+
+    public void processEventMessageV2(HmcMessage hmcMessage)
+        throws CaseException, MessageProcessingException {
+
+        Long caseId = hmcMessage.getCaseId();
+        String hearingId = hmcMessage.getHearingId();
+
+        HearingGetResponse hearingResponse = hmcHearingApiService.getHearingRequest(hearingId);
+
+        HmcStatus hmcMessageStatus = hmcMessage.getHearingUpdate().getHmcStatus();
+
+        if (stateNotHandled(hmcMessageStatus, hearingResponse)) {
+            logStateNotHandled(hearingId, caseId);
+            return;
+        }
+
+        logProcessingHmcMessage(hmcMessageStatus, hearingResponse, hearingId, caseId);
+
+        Function<SscsCaseDetails, Optional<DynamicEventUpdateResult>> mutator = sscsCaseDetails -> {
+
+            SscsCaseData sscsCaseData = sscsCaseDetails.getData();
+
+            try {
+                updateCaseDataMutator(sscsCaseData, hmcMessageStatus, hearingResponse, hearingId);
+            } catch (MessageProcessingException e) {
+                throw new RuntimeException(e);
+            } catch (InvalidMappingException e) {
+                throw new RuntimeException(e);
+            }
+
+            return resolveEventType(sscsCaseData, hmcMessageStatus, hearingResponse, hearingId);
+
+        };
+
+        IdamTokens idamTokens = idamService.getIdamTokens();
+
+        updateCcdCaseService.updateCaseV2DynamicEvent(caseId, idamTokens, mutator);
```

Comment:
I suppose there is no point in retrying if the case has been updated.

Reply:
I don't think a retry is needed here as there is one on the updateCaseV2DynamicEvent but we can discuss with the team